script.js

```diff
@@ -0,0 +1,42 @@
+function initHistogram() {
+	const dataInput = document.querySelector('#dataInput').value;
+	const validInputNumbers = validateInput(dataInput);
+
+	if (validInputNumbers.length === 0) {
+		alert("Введите данные!")
+		return;
+	}
+
+	drawHistogram(validInputNumbers);
+}
+
+function validateInput(input) {
+	return input
+		.split(' ')
+		.filter(item => !isNaN(Number(item)) && item !== '')
+}
+
+function createBarElement() {
+	const bar = document.createElement('div');
+	bar.classList.add('bar');
+	return bar;
+}
+
+function drawHistogram(dataArray) {
+	const maxWidth = (window.screen.width / 100) * dataArray.length;
+	const maxElement = Math.max(...dataArray);
+	const histogram = document.querySelector('.histogram');
+	histogram.innerHTML = '';
```

Comment:
Тут сразу два вопроса
1. Почему очистка делается в функции отрисовки гистограммы?
2. Почему это делается уже после того, как будет посчитана ширина, найден максимальный элемент?

Reply:
1) Чтобы не прописывать `const histogram = document.querySelector('.histogram');` в `initHistogram`, так как это уже есть в функции отрисовки
2) Не обратил внимания. Исправил

Reply:
Как вариант, можно очистку вынести в отдельную функцию.
Еще расскажи, как можно иначе сделать очистку? И можно ли?

Reply:
Можно использовать `removeChild` или `innerText`

Reply:
ещё есть функция `remove()`

Reply:
Сделал отдельную функцию и вызываю её в `initHistogram`
```suggestion
	function clearHistogram() {
	const histogram = document.querySelector('.histogram');
	histogram.innerHTML = '';
}
```